clis/teliod/src/command_listener.rs

```diff
@@ -0,0 +1,200 @@
+                    // send the response to teliod client
+                    connection.respond(response.serialize()).await?;
+                }
+            }
+        }
+        Ok(())
+    }
+
+    pub async fn handle_client_connection(&mut self) -> Result<ClientCmd, TeliodError> {
+        let mut connection = self.socket.accept().await?;
+        let command_str = connection.read_command().await?;
+
+        if let Ok(command) = serde_json::from_str::<ClientCmd>(&command_str) {
+            self.process_command(&command, &mut connection).await?;
+            return Ok(command);
+        } else {
+            error!("Received invalid command from client: {}", command_str);
+            connection
+                .respond(
+                    CommandResponse::Err(format!("Invalid command: {}", command_str)).serialize(),
+                )
+                .await?;
+            return Err(TeliodError::InvalidCommand(command_str));
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{ClientCmd, CommandResponse, DaemonSocket};
+    use serial_test::serial;
+    use std::path::Path;
+    use tokio::{
+        io::{AsyncReadExt, AsyncWriteExt},
+        net::UnixStream,
+        sync::mpsc,
+    };
+
+    const TEST_SOCKET_PATH: &str = "test_socket";
+
+    // Helper to create a fake command listener
+    fn make_command_listener() -> CommandListener {
+        let (tx, _rx) = mpsc::channel(1);
+        let (_tx, rx) = mpsc::channel(1);
+
+        let socket = DaemonSocket::new(Path::new(TEST_SOCKET_PATH)).unwrap();
+
+        CommandListener::new(socket, tx.clone(), rx)
+    }
+
+    // Simulate client sending command and waiting for response
+    async fn client_send_command(cmd: &str) -> Result<CommandResponse, std::io::Error> {
+        let mut client_stream = UnixStream::connect(&Path::new(TEST_SOCKET_PATH)).await?;
+        client_stream.write(format!("{}\n", cmd).as_bytes()).await?;
+
+        let mut data = vec![0; 1024];
+        let size = client_stream.read(&mut data).await?;
+        let response = String::from_utf8(data[..size].to_vec()).unwrap();
+        Ok(CommandResponse::deserialize(response.trim()).unwrap())
+    }
+
+    // Broken client, closes connection without waiting for response
+    async fn broken_client_send_command(cmd: &str) -> std::io::Result<()> {
+        let mut client_stream = UnixStream::connect(&Path::new(TEST_SOCKET_PATH)).await?;
+        client_stream.write(format!("{}\n", cmd).as_bytes()).await?;
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_command_response_serialization() {
+        let response = CommandResponse::Ok;
+        assert_eq!(response.serialize(), "\"Ok\"");
+
+        let error_response = CommandResponse::Err("Test error".to_string());
+        assert_eq!(error_response.serialize(), "{\"Err\":\"Test error\"}");
+    }
+
+    #[tokio::test]
+    #[serial]
```

Comment:
Any reason not to just create unix socket in temp dir with some random key, vs making all tests rerial?

Reply:
Yeah that's probably a better approach

Reply:
Fixed by adding current time `nanoseconds` to the socket name